fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -195,4 +223,243 @@ public void getTriplesOfBinaryDescription() throws Exception {
         assertEquals(dcTitleTriple, retrievedUserStream.findFirst().get());
     }
 
+    @Test
+    public void persistFailsIfCommitAlreadyComplete() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn("ocfl-object-id");
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        //perform the create rdf operation
+        session.persist(rdfSourceOperation);
+
+        //commit to OCFL
+        session.commit(CommitOption.UNVERSIONED);
+
+        //this should fail
+        try {
+            session.persist(rdfSourceOperation);
```

Comment:
Do `@Test(expected = PersistentStorageException.class)` instead of the catch

Reply:
I didn't use `expected` because if for some reason the previous persist() or commit() invocations throw a PersistentStorageException,  we wouldn't want the test to pass.

Reply:
Okay, that makes sense